src/core/mac/data_poll_sender.cpp

```diff
@@ -317,6 +317,12 @@ void DataPollSender::CheckFramePending(Mac::RxFrame &aFrame)
     {
```

Comment:
Here, for OT1.1 devices, it will send out data poll whenever it received a frame with frame pending set. This means if peer OT1.2 parent send out ACK with frame pending set in response to MAC data frame, OT1.1 child will follow up with a data poll which is against spec.

Reply:
Sending an IEEE 802.15.4 Data Request message in response to receiving an IEEE 802.15.4 ACK with Frame Pending = 1 is intentional here to reduce latency of transferring messages. I don't think the Thread 1.1 and 1.2 specifications prohibit this behavior. If you believe this behavior is not conformant, can you provide references to the relevant portions of the spec?

Reply:
Yes, for Thread 1.1, we want to ignore the Frame Pending bit in ACKs that are **not** in response to IEEE 802.15.4 Data Request frames.

Reply:
@brianljt 
I think the OT1.1 behavior with this PR is the same as before this PR. Note that only 1.2 will pass AckFrame to `CheckFramePending()`.
https://github.com/openthread/openthread/pull/4334/files#diff-073e11a5a34cdfd560305449d7763fe3R1306

Anything I missed regarding 1.1 behavior?

Reply:
Just adding a quick note. I think existing code (master) and 1.1 behavior is correct
- On master  `CheckFramePending()` is only called from rx path from `HandleReceivedFrame()`
- Received acks won't go through that path and are processed from tx callback `Mac::HandleTransmitDone()` 
- A pending bit in a received ack frame is only checked if the tx frame is data poll.

Reply:
Thank you @abtink @librasungirl for pointing this out. Then I think it is fine for OT1.1.